src/components/core/validateTransaction.ts

```diff
@@ -1,13 +1,44 @@
-import { JsonRpcProvider, Contract, Interface } from 'ethers'
+import { JsonRpcProvider, Contract, Interface, TransactionReceipt } from 'ethers'
 import { fetchEventFromTransaction } from '../../utils/util.js'
 import ERC20Template from '@oceanprotocol/contracts/artifacts/contracts/templates/ERC20TemplateEnterprise.sol/ERC20TemplateEnterprise.json' assert { type: 'json' }
 import ERC721Template from '@oceanprotocol/contracts/artifacts/contracts/templates/ERC721Template.sol/ERC721Template.json' assert { type: 'json' }
+import { CORE_LOGGER } from '../../utils/logging/common.js'
 
 interface ValidateTransactionResponse {
   isValid: boolean
   message: string
 }
 
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
+
+async function fetchTransactionReceipt(
+  txId: string,
+  provider: JsonRpcProvider,
+  retries: number = 3
+): Promise<TransactionReceipt> {
+  while (retries > 0) {
+    try {
+      CORE_LOGGER.logMessage(`fetching tx hash retry ${retries}....`)
+      const txReceipt = await provider.getTransactionReceipt(txId)
+      CORE_LOGGER.logMessage(`txReceipt: ${JSON.stringify(txReceipt)}`)
+      if (txReceipt) {
+        return txReceipt
+      }
+      CORE_LOGGER.logMessage(
+        `txReceipt not found #${retries}: ${JSON.stringify(txReceipt)}`
+      )
```

Comment:
To avoid creating a lot of noise in the logs when the retries number is set high, it would be nice to only log this message when we are on the final try (and then the other logs below can also be removed). Also, there is no point logging the `txReceipt` if one hasn't been recieved. For example: 

```suggestion
    retries--
    if(retries === 0){
    CORE_LOGGER.logMessage(
        `txReceipt not found after #${retries} tries.`
      )
    }
```